src/Microsoft.DotNet.Build.Tasks/PackageFiles/BuildVersion.targets

```diff
@@ -0,0 +1,15 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```

Comment:
> Is there any reason to not do this work directly in version.targets

Yes, We want this file to be one of the first imports in dir.props on the root of each repo, and by this it would set the right buildnumbermajor before other target files are loaded. That is why I have that as a separate file now.

> Or move the build number generation targets into this file as well?

I can do this if you prefer, the reason why I have it as it is now, is because I thought that everything that actually calculates what the version is should go under versioning.targets, whereas the actual file that needs to be imported first to calculate all of this needs to be separate to be loaded before other properties are set.